akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesShardStore.scala

```diff
@@ -138,78 +142,82 @@ private[akka] final class DDataRememberEntitiesShardStore(
       case GetDataDeleted(_, _) =>
         log.error("Unable to get an initial state because it was deleted")
         context.stop(self)
+      case update: RememberEntitiesShardStore.Update =>
+        log.warning("Got an update before load of initial entities completed, dropping update: [{}]", update)
     }
   }
 
-  private def addEntities(ids: Set[EntityId]): Unit = {
-    val updates: Map[Set[EntityId], (Update[ORSet[EntityId]], Int)] = ids.groupBy(key).map {
-      case (key, ids) =>
-        (ids, (Update(key, ORSet.empty[EntityId], writeMajority, Some(ids)) { existing =>
-          ids.foldLeft(existing) {
-            case (acc, nextId) => acc :+ nextId
-          }
-        }, maxUpdateAttempts))
-    }
+  private def onUpdate(update: RememberEntitiesShardStore.Update): Unit = {
+    // FIXME what about ordering of adds/removes vs sets, I think we can lose one
```

Comment:
What scenario is this? When we remove but the add hasn't been seen by this node yet?
Entity E1 in shard S1 on node A, adds
Shard S1 is moved to node B
Entity E1 is removed, but it  is on node B that hasn't seen the add on E1 yet 

Can this really happen? Either when Shard S1 starts on the other node it has seen E1 and starts it. Or it hasn't and a message arrives which would trigger S1 to add it before it removed it

Or is there another scenario?

Reply:
Yeah, perhaps it is protected by the state machine now. We don't allow transition between RememberingStart <-> RememberingStop so it would at least throw if it happens.

I'll remove the comment.